src/io/flutter/actions/InstallSdkAction.java

```diff
@@ -31,15 +53,166 @@ String getLinkText() {
     }
   }
 
+  private static class GitCloneAction extends InstallAction {
+    private final FlutterGeneratorPeer myPeer;
+
+    public GitCloneAction(FlutterGeneratorPeer peer) {
+      myPeer = peer;
+    }
+
+    @Override
+    void perform() {
+      @SuppressWarnings("DialogTitleCapitalization") final FileChooserDescriptor descriptor =
+        new FileChooserDescriptor(FileChooserDescriptorFactory.createSingleFolderDescriptor()) {
+          @Override
+          public void validateSelectedFiles(VirtualFile[] files) throws Exception {
+            for (VirtualFile file : files) {
+              if (file.findChild("flutter") != null) {
+                throw new IllegalArgumentException("A file callled 'flutter' already exists in this location.");
+              }
+            }
+          }
+        }.withTitle("Flutter SDK Directory").withDescription("Choose a directory to install the Flutter SDK");
+
+      final VirtualFile installTarget = FileChooser.chooseFile(descriptor, null, null);
+      if (installTarget != null) {
+        installTo(installTarget);
+      }
+    }
+
+    private void installTo(final @NotNull VirtualFile directory)  {
+      final String installPath = directory.getPath();
+
+      final GeneralCommandLine cmd = new GeneralCommandLine().withParentEnvironmentType(
+        GeneralCommandLine.ParentEnvironmentType.CONSOLE).withWorkDirectory(installPath).withExePath("git")
+        .withParameters("clone", "https://github.com/flutter/flutter.git");
+      runCommand(cmd, new CommandListener("Cloning Flutter repository…") {
+        @Override
+        void onError(@NotNull ExecutionException exception) {
+          showError("Error Cloning Flutter Repository", exception.getMessage());
+        }
+
+        @Override
+        protected void onSuccess(@NotNull ProcessEvent event) {
+          final String sdkDir = new File(directory.getPath(), "flutter").getPath();
+
+          final GeneralCommandLine cmd = new GeneralCommandLine().withParentEnvironmentType(
+            GeneralCommandLine.ParentEnvironmentType.CONSOLE).withWorkDirectory(sdkDir).withExePath("bin/flutter")
+            .withParameters("doctor");
+          runCommand(cmd, new CommandListener("Running Flutter doctor…") {
+            @Override
+            void onTextAvailable(ProcessEvent event, Key outputType) {
+              //TODO(pq): filter less useful / truncated messages.
```

Comment:
Unfortunately, there's still some bits we want to drop.

![image](https://user-images.githubusercontent.com/67586/27451119-e7e47464-5742-11e7-83b9-f96f7186cc01.png)


Will noodle on this in a follow-up.